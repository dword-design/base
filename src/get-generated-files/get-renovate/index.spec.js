import tester from '@dword-design/tester';
import testerPluginTmpDir from '@dword-design/tester-plugin-tmp-dir';
import { execa, execaCommand } from 'execa';
import fs from 'fs-extra';

import { Base } from '@/src/index.js';

export default tester(
  {
    'custom config': () =>
      expect(
        new Base({ renovateConfig: { foo: 'bar' } }).getRenovateConfig().foo,
      ).toEqual('bar'),
    'custom config array': () =>
      expect(
        new Base({ renovateConfig: { labels: ['foo'] } }).getRenovateConfig()
          .labels,
      ).toEqual(['maintenance', 'foo']),
    'github action': async () => {
      await execaCommand('git init');
      await execaCommand('git config user.email "foo@bar.de"');
      await execaCommand('git config user.name "foo"');
      await fs.outputFile('index.js', 'gitHubAction`actions/checkout@v3`');
      const base = new Base();
      await base.prepare();
      await execaCommand('git add .');
      await execa('git', ['commit', '-m', 'feat: init']);

      const output = (
        await execaCommand(
          `renovate --platform=local --dry-run --host-rules=[{matchHost:'api.github.com',token:'not-empty',hostType:'github'}]`,
          { env: { LOG_LEVEL: 'debug' } },
        )
      ).stdout;

      expect(output).toMatch(/renovate\/actions-checkout-\d+\.x/);
    },
    'lock file': async () => {
      await execaCommand('git init');
      await execaCommand('git config user.email "foo@bar.de"');
      await execaCommand('git config user.name "foo"');

      await fs.outputFile(
        'package.json',
        JSON.stringify({ dependencies: { globby: '^13.0.0' } }),
      );

      await execaCommand('pnpm install');
      const base = new Base();
      await base.prepare();
      await execaCommand('git add .');
      await execa('git', ['commit', '-m', 'feat: init']);

      const output = (
        await execaCommand('renovate --platform=local --dry-run', {
          env: { LOG_LEVEL: 'debug' },
        })
      ).stdout;

      expect(output).toMatch('chore: lock file maintenance');
    },
    'lock file fix commit type': async () => {
      await execaCommand('git init');
      await execaCommand('git config user.email "foo@bar.de"');
      await execaCommand('git config user.name "foo"');

      await fs.outputFile(
        'package.json',
        JSON.stringify({ dependencies: { globby: '^13.0.0' } }),
      );

      await execaCommand('pnpm install');
      const base = new Base({ isLockFileFixCommitType: true });
      await base.prepare();
      await execaCommand('git add .');
      await execa('git', ['commit', '-m', 'feat: init']);

      const output = (
        await execaCommand('renovate --platform=local --dry-run', {
          env: { LOG_LEVEL: 'debug' },
        })
      ).stdout;

      expect(output).toMatch('fix: lock file maintenance');
    },
    'nodejs version': async () => {
      await execaCommand('git init');
      await execaCommand('git config user.email "foo@bar.de"');
      await execaCommand('git config user.name "foo"');
      await fs.outputFile('index.js', 'nodejsVersion`18`');
      const base = new Base();
      await base.prepare();
      await execaCommand('git add .');
      await execa('git', ['commit', '-m', 'feat: init']);

      const output = (
        await execaCommand('renovate --platform=local --dry-run', {
          env: { LOG_LEVEL: 'debug' },
        })
      ).stdout;

      expect(output).toMatch(/renovate\/node-\d+\.x/);
    },
    'outdated version in package.json': async () => {
      await execaCommand('git init');
      await execaCommand('git config user.email "foo@bar.de"');
      await execaCommand('git config user.name "foo"');

      await fs.outputFile(
        'package.json',
        JSON.stringify({ dependencies: { globby: '^13.0.0' } }),
      );

      const base = new Base();
      await base.prepare();
      await execaCommand('git add .');
      await execa('git', ['commit', '-m', 'feat: init']);

      const output = (
        await execaCommand('renovate --platform=local', {
          env: { LOG_LEVEL: 'debug' },
        })
      ).stdout;

      expect(output).toMatch(/renovate\/globby-\d+\.x/);
    },
  },
  [testerPluginTmpDir()],
);
