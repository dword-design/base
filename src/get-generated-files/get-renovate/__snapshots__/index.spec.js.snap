// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index base 1`] = `
Object {
  "extends": Array [
    ":semanticCommits",
    ":semanticPrefixFix",
  ],
  "github-actions": Object {
    "enabled": false,
  },
  "labels": Array [
    "maintenance",
  ],
  "lockFileMaintenance": Object {
    "automerge": true,
    "enabled": true,
    "semanticCommitType": "chore",
  },
  "rangeStrategy": "replace",
  "regexManagers": Array [
    Object {
      "datasourceTemplate": "github-tags",
      "fileMatch": Array [
        "\\\\.js$",
      ],
      "matchStrings": Array [
        "(^|\\\\s)gitHubAction\`(?<depName>.*?)@v(?<currentValue>.*?)\`",
      ],
      "versioningTemplate": "npm",
    },
  ],
  "semanticCommitScope": null,
}
`;

exports[`index lock file commit type 1`] = `
Object {
  "extends": Array [
    ":semanticCommits",
  ],
  "ignorePaths": Array [
    ".github/workflows/build.yml",
  ],
  "labels": Array [
    "maintenance",
  ],
  "lockFileMaintenance": Object {
    "enabled": true,
  },
  "packageRules": Array [
    Object {
      "depTypeList": Array [
        "dependencies",
        "devDependencies",
        "peerDependencies",
        "optionalDependencies",
      ],
      "semanticCommitType": "fix",
    },
  ],
  "semanticCommitScope": null,
}
`;

exports[`index lock file fix commit type 1`] = `
Object {
  "extends": Array [
    ":semanticCommits",
    ":semanticPrefixFix",
  ],
  "github-actions": Object {
    "enabled": false,
  },
  "labels": Array [
    "maintenance",
  ],
  "lockFileMaintenance": Object {
    "automerge": true,
    "enabled": true,
  },
  "rangeStrategy": "replace",
  "regexManagers": Array [
    Object {
      "datasourceTemplate": "github-tags",
      "fileMatch": Array [
        "\\\\.js$",
      ],
      "matchStrings": Array [
        "(^|\\\\s)gitHubAction\`(?<depName>.*?)@v(?<currentValue>.*?)\`",
      ],
      "versioningTemplate": "npm",
    },
  ],
  "semanticCommitScope": null,
}
`;

exports[`index not base 1`] = `
Object {
  "extends": Array [
    ":semanticCommits",
    ":semanticPrefixFix",
  ],
  "github-actions": Object {
    "enabled": false,
  },
  "labels": Array [
    "maintenance",
  ],
  "lockFileMaintenance": Object {
    "automerge": true,
    "enabled": true,
    "semanticCommitType": "chore",
  },
  "rangeStrategy": "replace",
  "regexManagers": Array [
    Object {
      "datasourceTemplate": "github-tags",
      "fileMatch": Array [
        "\\\\.js$",
      ],
      "matchStrings": Array [
        "(^|\\\\s)gitHubAction\`(?<depName>.*?)@v(?<currentValue>.*?)\`",
      ],
      "versioningTemplate": "npm",
    },
  ],
  "semanticCommitScope": null,
}
`;
